dali/operators/image/convolution/laplacian.cc

```diff
@@ -0,0 +1,222 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "dali/core/static_switch.h"
+#include "dali/kernels/imgproc/convolution/laplacian_cpu.h"
+#include "dali/kernels/kernel_manager.h"
+#include "dali/operators/image/convolution/laplacian.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+using namespace convolution_utils;  // NOLINT
+
+DALI_SCHEMA(Laplacian)
+    .DocStr(R"code(Computes laplacian of the input.
+
+Laplacian is calculated as a sum of second order partial derivatives in each spatial
+dimension of the data. Each partial derivative is approximated with a separable convolution,
+that uses derivative window in the direction of the partial derivative and smoothing windows in
+the remaining axes.
+
+In total, for ``n`` spatial dimensions, there are ``n * n`` window sizes.
+The window size parameter can be:
+  * A single value used for all windows.
+  * ``n * n`` values, where ``window_size[i][j]`` is a size of window applied along ``j``-th axis
+    while computing ``i``-th partial derivative. Values ``window_size[i][i]`` describe
+    derivative window sizes, the remianing values refer to smoothing windows.
+  * ``n`` values, where the first value describes the derivative window size and
+    remaining values refer to smoothing windows. Values are shifted to the right to
+    obtain parameters for all partial derivatives. For instance, for volumetric data, specifing
+    ``window_sizes = [w1, w2, w3]`` is same as setting
+    ``window_sizes = [w1, w2, w3, w3, w1, w2, w2, w3, w1]``.
+
+Window sizes must be odd. Size of a derivative window must be at least 3. Smoothing
+window can be of size 1, which implies no smoothing along corresponding axis.
+
+To normalize output, derivative kernel that uses ``n`` windows of total size equal to ``s``
+should be scaled by 2^(-s + n + 2). If ``normalize`` argument is set to True,
+normalization is done by the operator. Alternatively, you can specify ``scale`` argument
+to customize scaling factors. Scale can be either a single value or ``n`` values, one for every
+partial derivative.
+
+Operator uses floats as an intermediate type.
+
+.. note::
+  The channel ``C`` and frame ``F`` dimensions are not considered data axes. If channels are present,
+  only channel-first or channel-last inputs are supported.
+
+)code")
+    .NumInput(1)
+    .NumOutput(1)
+    .AllowSequences()
+    .SupportVolumetric()
+    .AddOptionalArg<int>(laplacian::kWindowSizeArgName,
+                         "Sizes of windows used in convolutions",
+                         std::vector<int>{laplacian::defaultWindowSize}, true)
+    .AddOptionalArg<float>(laplacian::scaleArgName,
```

Comment:
Shouldn't this be fully optional? (pass a nullptr here)?

Reply:
If you don't specify neither scale nor normalize, it will use scale of size 1 so it seemed like a good default. Maybe the weird part is that if you use normalize=True this default is ignored, but is it misleading?

Reply:
It's a bit misleading, when we say, that setting normalize=True for scale that was set is an error, and the scale is set by default. Unless we already do it in other operators (we probably do), but there was a reason to introduce the empty default to not have those weird cases. 

How common is no-normalization vs normalization? Which would be more sensible default?

Reply:
It's an error to specify both `normalize` and `scale` but only if `scale` is set explicitly. The default 1 is ignored if you specify true for `normalize`.

As to which one is more common - I really don't know. The fact is that OpenCV does not provide normalization but rather `scale` arg with a reference to the function that returns kernels where formula for scaling factor is described in the docs.